test/e2e/create_test.go

```diff
@@ -44,50 +45,114 @@ func testWindowsNodeCreation(t *testing.T) {
 	// Create a private key secret with the known private key.
 	require.NoError(t, testCtx.createPrivateKeySecret(true), "could not create known private key secret")
 
-	t.Run("Windows Machines without the Windows label are not configured", func(t *testing.T) {
-		// Test is platform agnostic so is not needed to be run for every supported platform.
-		if testCtx.CloudProvider.GetType() != config.AzurePlatformType {
-			t.Skipf("Skipping for %s", testCtx.CloudProvider.GetType())
+	t.Run("Machine controller", testCtx.testMachineConfiguration)
+	// BYOH creation must occur after the Machine creation, as the MachineConfiguration tests change the private key
+	// multiple times, and BYOH doesnt have the functionality of rotating keys on the VMs. This would result in BYOH
+	// failing the pub key annotation validation as it compares the current private key secret with the annotation.
+	t.Run("ConfigMap controller", testCtx.testBYOHConfiguration)
+
+}
+
+// ensureWindowsInstanceConfigMapDeleted deletes the windows-instances configmap if it exists
+func (tc *testContext) ensureWindowsInstanceConfigMapDeleted() error {
+	err := tc.client.K8s.CoreV1().ConfigMaps(tc.namespace).Delete(context.TODO(), controllers.InstanceConfigMap, metav1.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	return nil
+}
+
+// createWindowsInstanceConfigMap creates a ConfigMap for the ConfigMap controller to act on, comprised of the Machines
+// in the given MachineList
+func (tc *testContext) createWindowsInstanceConfigMap(machines *mapi.MachineList) error {
+	var username string
+	if tc.CloudProvider.GetType() == config.AzurePlatformType {
+		username = "capi"
+	} else {
+		username = "Administrator"
+	}
+
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: controllers.InstanceConfigMap,
+		},
+		Data: make(map[string]string),
+	}
+	for _, machine := range machines.Items {
+		for _, address := range machine.Status.Addresses {
+			if address.Type == v1.NodeInternalIP || address.Type == v1.NodeInternalDNS {
+				if tc.CloudProvider.GetType() == config.VSpherePlatformType && address.Type == v1.NodeInternalDNS {
+					// skip using DNS on vSphere as it is not working
```

Comment:
If you know this for a fact, please bring this up with the SPLAT team.

Reply:
I dont actually think this is the case, but instead the issue is that the Machine does not have an IP yet even though its in the provisioned state. Making that check in the waitForWindowsMachines function instead